src/stores/v2/derived.ts

```diff
@@ -76,15 +76,21 @@ export const vaultsAggregatedCoveredDebt = derived([vaultsStore, balancesStore],
     let initialVal = BigNumber.from(0);
     balances = {
       ...balances,
-      [Number(vaultTypeKey)]: vaultBody.reduce((_prevVault, _currentVault) => {
-        const underlyingTokenData = getTokenDataFromBalances(_currentVault.underlyingAddress, [$balances]);
-        const normalizedAmount = _currentVault.balance
-          .mul(_currentVault.underlyingPerShare)
-          .div(BigNumber.from(10).pow(underlyingTokenData.decimals))
-          .mul(BigNumber.from(10).pow(BigNumber.from(18).sub(underlyingTokenData.decimals)));
-
-        return _prevVault.add(normalizedAmount);
-      }, initialVal),
+      [Number(vaultTypeKey)]:
+        (vaultBody)
```

Comment:
I'm actually unsure if this check is necessary; line 68 already checks for `$vaults` so the const `{vaultBody}` should never be undefined.

Reply:
Could've sworn I saw this throwing an error at some point, but it looks like you're right. I reverted the code and ran tests on all button/toggle combinations for vault items and saw no issues. I've made a new push that reverts this.